sensor/tests/resource/helper.go

```diff
@@ -229,6 +233,80 @@ func runPermutation(files []YamlTestFile, i int, cb func([]YamlTestFile)) {
 	}
 }
 
+// AssertFunc is the deployment state assertion function signature.
+type AssertFunc func(deployment *storage.Deployment) bool
```

Comment:
Maybe it would be better to pass here `testing.T` to be able to assert inside or maybe return an error? 🤔  For these tests this is ok since there's only one assertion but in the service tests there are multiple assertions for each deployment and there is no way to know which assertion failed if we return a `bool`

Reply:
Yea I thought about that. But then the test would fail on the first check of the state. The reason why I explicitly didn't want to use a test assertion here is because the semantic of this function is "eventually this assertion should be true". If Go test has a single `assert.Equal` that fails the complete test fails -- even if the most recent deployment/violation state passes the assertion. 

If you have a suggestion of how to workaround that or design in a way that we can assert multiple state but only consider the last assertion as the test result, I'd be happy to change this.

Reply:
> maybe return an error?

Returning an error might be an improvement. If there are no other ideas of how to restructure this I'll change it to return an error.

Reply:
That makes sense. Then maybe returning an `error` or allowing the `AssertFunc` to modify the `message`.